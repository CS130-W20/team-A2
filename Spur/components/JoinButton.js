import React, {Component} from 'react';
import Icon from 'react-native-vector-icons/FontAwesome';
import DatabaseManager from '../classes/DatabaseManager'; 
import { Image,
	 Platform,
	 StyleSheet,
	 Text,
	 TextInput,
	 TouchableOpacity,
     Alert } from 'react-native';
import {Button} from 'react-native-elements';


export default class JoinButton extends Component<Props>
{
    constructor(props) {
		super(props); 
		//Setup firebase via a databaseManager
		this.databaseManager = new DatabaseManager();
		this.state = {
            event: this.props.event,
            isAttendee: this.props.isAttendee,
            isCheckedIn: this.props.isCheckedIn,
        }
        
    }
    
    /** React render function **/
    render() {
        
        this.databaseManager = new DatabaseManager();
        const eventId = this.props.eventId;
        const event = this.state.event;
        const uid = this.props.uid;
        const upcoming = this.props.upcoming;

        if (!this.state.isAttendee) {
            return( <Button
                title = 'Join Event!'
                onPress={() => 
                    {
                        
                        if (event.attendees.indexOf(uid) < 0) {
                            event.attendees.push(uid);

                            if (upcoming.indexOf(eventId) < 0) {
                                upcoming.push(eventId);
                            }
                            this.databaseManager.updateEvent(eventId, event);
                            this.databaseManager.updateUser(uid, {upcoming: upcoming});
                        }
                    
                        Alert.alert('Event Joined!');
                        
                        this.setState({isAttendee: true, isCheckedIn: false});
                    }
                }
            />);
        } else {

            if (!this.state.isCheckedIn) {
                return( <Button
                    title = 'Check into Event!'
                    onPress={() => 
                        {
                            const checkedIn = event.checked_in ? event.checked_in : []

                            if (checkedIn.indexOf(uid) < 0) {
                                checkedIn.push(uid);
                                this.databaseManager.updateEvent(eventId, {checked_in: checkedIn});
                            }
                           
                            
                            //Alert.alert('Will check user into the event');
                            this.setState({isAttendee: true, isCheckedIn: true});
                        }
                    }
                />);
            } else {
                return( <Button
                    title = "You're checked in!"
                    icon={
                        <Icon
                          name="check"
                          size={15}
                          color="white"
                        />
                      }
                    color="green"
                />);
            }
        }
    }


}
