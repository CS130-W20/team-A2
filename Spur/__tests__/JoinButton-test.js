import * as React from 'react';
import renderer from 'react-test-renderer';
import JoinButton from '../components/JoinButton';

describe(`Join Button Tester`, () => {
    test('Join Button test', () => {
    
        const event = {};

        const isAttendee = false;
        const isCheckedIn = false;
        const eventId = '12345';
        const uid = '12345';
        const userName = 'Greg';
        const upcoming = ['1234'];

        const button = renderer.create(<JoinButton 
            event={event} 
            isAttendee={isAttendee} 
            isCheckedIn={isCheckedIn} 
            eventId={eventId}
            uid={uid}
            userName={userName}
            upcoming={upcoming}
            
            />).toJSON();
        expect(button).toMatchSnapshot();

    });

    test('Checkin Button test', () => {
    
        const event = {};

        const isAttendee = true;
        const isCheckedIn = false;
        const eventId = '12345';
        const uid = '12345';
        const userName = 'Greg';
        const upcoming = ['1234'];

        const button = renderer.create(<JoinButton 
            event={event} 
            isAttendee={isAttendee} 
            isCheckedIn={isCheckedIn} 
            eventId={eventId}
            uid={uid}
            userName={userName}
            upcoming={upcoming}
            
            />).toJSON();
        expect(button).toMatchSnapshot();

    });

    test('Checkin Time Invalid test', () => {
    
        const event = {};

        const isAttendee = true;
        const isCheckedIn = false;
        const eventId = '12345';
        const uid = '12345';
        const userName = 'Greg';
        const upcoming = ['1234'];
        const startTime = {
            hours: 24,
            minutes: 60
        }

        const button = renderer.create(<JoinButton 
            event={event} 
            isAttendee={isAttendee} 
            isCheckedIn={isCheckedIn} 
            eventId={eventId}
            uid={uid}
            userName={userName}
            upcoming={upcoming}
            
            />).toJSON();
        expect(button).toMatchSnapshot();

    });

    test('Checked In Test', () => {
    
        const event = {};

        const isAttendee = true;
        const isCheckedIn = true;
        const eventId = '12345';
        const uid = '12345';
        const userName = 'Greg';
        const upcoming = ['1234'];

        const button = renderer.create(<JoinButton 
            event={event} 
            isAttendee={isAttendee} 
            isCheckedIn={isCheckedIn} 
            eventId={eventId}
            uid={uid}
            userName={userName}
            upcoming={upcoming}
            
            />).toJSON();
        expect(button).toMatchSnapshot();

    });

});