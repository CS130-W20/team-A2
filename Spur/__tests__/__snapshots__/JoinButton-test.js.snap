// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join Button Tester Checked In Test 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2089dc",
          "borderColor": "#2089dc",
          "borderRadius": 3,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
            "paddingBottom": 1,
            "paddingTop": 2,
            "textAlign": "center",
          }
        }
      >
        You're checked in!
      </Text>
    </View>
  </View>
</View>
`;

exports[`Join Button Tester Checkin Button test 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2089dc",
          "borderColor": "#2089dc",
          "borderRadius": 3,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
            "paddingBottom": 1,
            "paddingTop": 2,
            "textAlign": "center",
          }
        }
      >
        Check into Event!
      </Text>
    </View>
  </View>
</View>
`;

exports[`Join Button Tester Checkin Time Invalid test 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2089dc",
          "borderColor": "#2089dc",
          "borderRadius": 3,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
            "paddingBottom": 1,
            "paddingTop": 2,
            "textAlign": "center",
          }
        }
      >
        Check into Event!
      </Text>
    </View>
  </View>
</View>
`;

exports[`Join Button Tester Join Button test 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2089dc",
          "borderColor": "#2089dc",
          "borderRadius": 3,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
            "paddingBottom": 1,
            "paddingTop": 2,
            "textAlign": "center",
          }
        }
      >
        Join Event!
      </Text>
    </View>
  </View>
</View>
`;
